File: src/main/java/com/example/springtilcode/domain/user/controller/UserController.java
package com.example.springtilcode.domain.user.controller;

import com.example.springtilcode.domain.user.controller.dto.QuerydslDto;
import com.example.springtilcode.domain.user.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class UserController {

    private final UserService userService;

    @GetMapping("/user/{id}")
    public QuerydslDto getUserInfo(@PathVariable long id){
        return userService.findUserInfo(id);
    }
}

File: kotlin/src/main/kotlin/com/example/demo/domain/user/api/UserController.kt
package com.example.demo.domain.user.api

import com.example.demo.domain.user.dto.request.EditProfileRequest
import com.example.demo.domain.user.dto.response.UserResponse
import com.example.demo.domain.user.application.UserService
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class UserController(
    val userService: UserService
) {
    @GetMapping("/user")
    fun getUser() : UserResponse {
        return userService.getUser()
    }

    @PatchMapping("/user")
    fun editProfile(@Valid @RequestBody request: EditProfileRequest) : UserResponse {
        return userService.editProfile(request.email)
    }
}